src/pip/_internal/commands/check.py

```diff
@@ -45,8 +50,13 @@ def run(self, options: Values, args: List[str]) -> int:
                     dep_name,
                     dep_version,
                 )
-
-        if missing or conflicting or parsing_probs:
+        for package in unsupported:
+            write_output(
+                "%s %s is not supported on this platform",
+                package.canonical_name,
```

Comment:
For user-facing messages `raw_name` is generally preferred.

Reply:
hyphen or space?